crates/walrus-service/src/node/config.rs

```diff
@@ -594,6 +598,33 @@ impl BlobRecoveryConfig {
     }
 }
 
+/// Configuration for the pending sliver cache.
+#[serde_as]
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+#[serde(default)]
+pub struct PendingSliverCacheConfig {
+    /// Maximum number of slivers retained in memory while the blob registration is pending.
+    pub max_cached_slivers: usize,
```

Comment:
AFAICT, this is also used to limit the entries in the metadata cache, which I find not very intuitive. 

Suggestion: Add a separate `max_metadata_cache_entries` (and use a default value which is smaller by a factor of ~10).